scripts/lib/CIME/SystemTests/erp.py

```diff
@@ -8,132 +8,72 @@
-                self._case.set_value("HIST_N", stop_n)
-                self._case.set_value("HIST_OPTION", stop_option)
-                self._case.set_value("CONTINUE_RUN", False)
-                suffix = "base"
-            else:
-                rest_n = stop_n/2 + 1
-                stop_new = stop_n - rest_n
-                expect(stop_new > 0, "ERROR: stop_n value {:d} too short {:d} {:d}".format(stop_new,stop_n,rest_n))
-                self._case.set_value("STOP_N", stop_new)
-                self._case.set_value("CONTINUE_RUN", True)
-                self._case.set_value("REST_OPTION","never")
-                suffix = "rest"
-
-            case_setup(self._case, test_mode=True, reset=True)
-
-            self.run_indv(suffix=suffix)
 
-        self._component_compare_test("base", "rest")
+    def _case_one_setup(self):
+        stop_n      = self._case.get_value("STOP_N")
+
+        expect(stop_n > 2, "ERROR: stop_n value {:d} too short".format(stop_n))
+
+    def _case_two_setup(self):
+        # halve the number of tasks and threads
+        for comp in self._case.get_values("COMP_CLASSES"):
+            ntasks    = self._case1.get_value("NTASKS_{}".format(comp))
+            nthreads  = self._case1.get_value("NTHRDS_{}".format(comp))
+            rootpe    = self._case1.get_value("ROOTPE_{}".format(comp))
+            if ( nthreads > 1 ):
+                self._case.set_value("NTHRDS_{}".format(comp), nthreads/2)
+            if ( ntasks > 1 ):
+                self._case.set_value("NTASKS_{}".format(comp), ntasks/2)
+                self._case.set_value("ROOTPE_{}".format(comp), rootpe/2)
+
+        stop_n = self._case1.get_value("STOP_N")
+        rest_n = self._case1.get_value("REST_N")
+        stop_new = stop_n - rest_n
+        expect(stop_new > 0, "ERROR: stop_n value {:d} too short {:d} {:d}".format(stop_new,stop_n,rest_n))
+        self._case.set_value("STOP_N", stop_new)
+        self._case.set_value("HIST_N", stop_n)
+        self._case.set_value("CONTINUE_RUN", True)
+        self._case.set_value("REST_OPTION","never")
+
+        # Note, some components, like CESM-CICE, have
+        # decomposition information in env_build.xml that
+        # needs to be regenerated for the above new tasks and thread counts
+        case_setup(self._case, test_mode=True, reset=True)
+
+    def _case_one_custom_postrun_action(self):
+        rundir1 = self._case1.get_value("RUNDIR")
+        rundir2 = self._case2.get_value("RUNDIR")
+        case = self._case1.get_value("CASE")
+        datenames = _get_datenames(self._case1)
+        for file_ in glob.iglob(os.path.join(rundir1,"*")):
+            logger.info("File is {}".format(file_))
+            if os.path.basename(file_).startswith("rpointer"):
+                logger.info("Copy {} to {}".format(file_, rundir2))
+                shutil.copy(file_, rundir2)
+            elif os.path.basename(file_).startswith(case) and datenames[0] in file_:
```

Comment:
Is this guaranteed to pick up everything that's needed? There's a lot of complexity in `_archive_restarts` that isn't captured here. For the most part, I don't see what this might be missing, except that it seems not to capture any unfinished history files that the short-term archiver would capture with `get_histfiles_for_restarts`.

(This is why I was hoping we could still reuse pieces of the case_st_archive code....)